internal/builder/builder_test.go

```diff
@@ -240,15 +240,15 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
 				})
 			})
 
-			when("missing stack id label", func() {
+			when("missing stack id label and run image", func() {
 				it.Before(func() {
 					h.AssertNil(t, baseImage.SetEnv("CNB_USER_ID", "1234"))
 					h.AssertNil(t, baseImage.SetEnv("CNB_GROUP_ID", "4321"))
 				})
 
-				it("returns an error", func() {
+				it("does not return an error", func() {
 					_, err := builder.New(baseImage, "some/builder")
-					h.AssertError(t, err, "image 'base/image' missing label 'io.buildpacks.stack.id'")
+					h.AssertNilE(t, err)
```

Comment:
Why don't we error here?

Reply:
Ah, makes sense - I guess what threw me was the addition of `and run image` to the test description. I thought of a builder that is missing a run image reference.